apps/dashboard/app/apps/request_tracker_client.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require 'rest_client'
+
+# HTTP client to create a request tracker ticket using the API
+# Configuration parameters:
+# - `server`: URL for the Request Tracker server (required)
+# - `user`: RT API username
+# - `pass`: RT API password
+# - `auth_token`: RT API [auth token](https://github.com/bestpractical/rt-authen-token), preferred instead of username and password.
+# - `timeout`: Connection and read timeout in seconds. Defaults to 30.
+# - `verify_ssl`: Whether or not the client should validate SSL certificates. Defaults to true.
+# - `proxy`: Proxy server URL. Defaults to no proxy.
+#
+class RequestTrackerClient
+  # THIS CLIENT IS BASED ON https://github.com/uidzip/rt-client
+
+  def self.create
+    rt_config = ::Configuration.support_ticket_config.fetch(:rt_api, {})
+
+    new({
+          server:     rt_config[:server],
+          user:       rt_config[:user],
+          pass:       rt_config[:pass],
+          auth_token: rt_config[:auth_token],
+          timeout:    rt_config[:timeout],
+          verify_ssl: rt_config[:verify_ssl],
+          proxy:      rt_config[:proxy]
+        })
+  end
+
+  UA = 'Open OnDemand ruby RT Client'
+  attr_reader :server, :rt_client, :resource, :timeout, :verify_ssl
+
+  def initialize(config)
+    # DEFAULTS
+    @auth_token = nil
+    @timeout = 30
+    @verify_ssl = false
+    # FROM CONFIGURATION
+    @user = config[:user]
+    @pass = config[:pass]
+    @auth_token = config[:auth_token]
+    @timeout = config[:timeout] if config[:timeout]
+    @verify_ssl = config[:verify_ssl] if config[:verify_ssl]
+    if config[:server]
+      @server = config[:server]
+      @server += '/' if @server !~ %r{/$}
+      @resource = "#{@server}REST/1.0/"
+    end
+
+    raise ArgumentError, 'server is a required option for RT client' unless @server
+
+    if !@auth_token && (!@user || !@pass)
+      raise ArgumentError, 'user and pass are required if not auth_token is provided for RT client'
+    end
+
+    if !@auth_token && !@user && !@pass
+      raise ArgumentError, 'auth_token or user and pass are required options for RT client'
+    end
+
+    headers = { 'User-Agent' => UA,
+                'Cookie'     => '' }
+    headers['Authorization'] = "Token #{@auth_token}" if @auth_token
+
+    options = {
+      headers:    headers,
+      timeout:    @timeout,
+      verify_ssl: @verify_ssl
+    }
+    options[:proxy] = config[:proxy] if config[:proxy]
+
+    @rt_client = RestClient::Resource.new(@resource, options)
+    untaint
```

Comment:
Why is this here?

Reply:
The client is based on the RT client library: https://github.com/uidzip/rt-client
`untaint` is originally in the lib client and I didn't remove or question its use.

I have removed from the code and it works as before.